src/Api/AdminConsole/Queries/OrganizationUserUserDetailsQuery.cs

```diff
@@ -0,0 +1,92 @@
+﻿using Bit.Core.Enums;
+using Bit.Core.Models.Data;
+using Bit.Core.Models.Data.Organizations.OrganizationUsers;
+using Bit.Core.Repositories;
+using Bit.Core.Services;
+using Bit.Core.Utilities;
+
+namespace Api.AdminConsole.Queries;
+
+// Request object. If we need to do the Auth check in the query (see below). 
+// The ClaimsPrincipal would need to be passed in from the endpoint. I see the
+// AuthorizeAsync sort of does the same thing with the ClaimsPrincipal for passing
+// to the service so it might not be a big deal
+public class OrganizationUserUserDetailsQueryRequest
+{
+    // Again if this is necessary. Wanted to name it something so 
+    // it's clear it is not the Bitwarden user info. Rather the ClaimsPrincipal
+    // public ClaimsPrincipal RequestingUserContext { get; set; }
+    public Guid OrganizationId { get; set; }
+    public bool IncludeGroups { get; set; } = false;
+    public bool IncludeCollections { get; set; } = false;
+}
+
+public class OrganizationUserUserDetailsQueryResponse
+{
+    public OrganizationUserUserDetails OrganizationUserUserDetails { get; set; }
+    public bool TwoFactorEnabled { get; set; }
+    public Permissions Permissions { get; set; }
+
+    public OrganizationUserUserDetailsQueryResponse(
+        OrganizationUserUserDetails organizationUserUserDetails,
+        bool twoFactorEnabled
+    )
+    {
+        this.OrganizationUserUserDetails = organizationUserUserDetails;
+        this.TwoFactorEnabled = twoFactorEnabled;
+        // Convert the OrganizationUserUserDetails permissions json string 
+        this.Permissions = CoreHelpers.LoadClassFromJsonData<Permissions>(OrganizationUserUserDetails.Permissions);
```

Comment:
The class already has a helper for this: `OrganizationUserUserDetails.GetPermissions()`. Although my preference is for the consuming code to do this as/when required, it's duplicative here.

Reply:
I don't know how I missed that. That is 100 times easier. Thanks.